service_container/calls.rst

```diff
@@ -71,3 +71,74 @@ To configure the container to call the ``setLogger`` method, use the ``calls`` k
 
         $container->register(MessageGenerator::class)
             ->addMethodCall('setLogger', [new Reference('logger')]);
+
+When it comes to optional dependencies, the setter approach can be useful but
+what if you need to change the injected dependency without mutating the initial service?
+
```

Comment:
I propose to add a versionadded directive here

Reply:
Just added a `versionadded` block, to be checked (not sure about the syntax :/ )